### PR TITLE
Add detection for CVE-2024-8956 and CVE-2024-8957

### PR DESCRIPTION
**Pull request by nmasdoufi-ol**
### Summary

This pull request introduces detection and exploitation support for two critical vulnerabilities in the ValueHD PTZ Camera firmware below version 6.3.40:

- **CVE-2024-8956**: Authentication Bypass
- **CVE-2024-8957**: Command Injection via NTP server configuration

Both vulnerabilities are cataloged as **CRITICAL** due to their potential impact on device security.

### Changes

1. **Detection Mechanisms**:
    - **Authentication Bypass**: Detects bypass vulnerability by checking for vulnerable firmware versions on the `/cgi-bin/param.cgi?get_device_conf` endpoint.
    - **Command Injection**: Attempts to execute a command by injecting code into the NTP configuration on the `/cgi-bin/param.cgi?post_network_other_conf` endpoint.
	- **Accept Pattern**: Matches against `versioninfo="SOC v\d+\.\d+\.\d+` to confirm firmware version compatibility.
	- **Version Validation**: Checks if the firmware version is below `6.3.40`, marking it as vulnerable if true.

2. **Metadata Adjustments**:
    - Added `reference` field to support both `str` and `list[str]` types for better handling of multiple CVE references.

3. **Enhancements in `webexploit.py`**:
    - Improved `_create_vulnerability()` method to support multiple references.
    - Added detailed NVD links for each CVE reference, enabling quick access to official CVE details.

image

### CODE REVIEW

File: agent/exploits/webexploit.py [hunk 1]
```diff
@@ -120,7 +120,8 @@ def _create_vulnerability(
             short_description=self.metadata.description,
             description=self.metadata.description,
             references={
-                "nvd.nist.gov": f"https://nvd.nist.gov/vuln/detail/{self.metadata.reference}",
+                f"nvd.nist.gov/{cve}": f"https://nvd.nist.gov/vuln/detail/{cve}"
+                for cve in self.metadata.reference
             },
```
Comment (PiranhaSa) [inline agent/exploits/webexploit.py:125]: You need to make sure it's a list not string

Comment (nmasdoufi-ol) [reply agent/exploits/webexploit.py]: Good catch.

Comment (ybadaoui-ostorlab) [reply agent/exploits/webexploit.py]: +1

File: agent/exploits/vhd_ptz_exploit.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        vulnerabilities: list[definitions.Vulnerability] = []
+
+        target_endpoint = self.inject_request.path
+        try:
+            resp = self.session.post(
+                f"{target.origin}{target_endpoint}",
+                data=self.inject_request.data,
+                timeout=DEFAULT_TIMEOUT.seconds,
+            )
+            if resp.status_code == 200 and "Success" in resp.text:
```
Comment (elyousfi5) [inline agent/exploits/vhd_ptz_exploit.py:99]: check on .lower()
```suggestion
            if resp.status_code == 200 and "Success" in resp.text:
```

Comment (nmasdoufi-ol) [reply agent/exploits/vhd_ptz_exploit.py:99]: No the response is like this: `{"Response":{"Result":"Success"}}`

File: agent/definitions.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ class VulnerabilityMetadata:
 
     title: str
     description: str
-    reference: str
+    reference: str | list[str]
```
Comment (3asm) [inline agent/definitions.py:56]: Sorry but this is not clean, we simply need to switch to references of list[str] only and changes all the other code.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/exploits/vhd_ptz_exploit.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
```
Comment (ohachimOs) [inline agent/exploits/vhd_ptz_exploit.py:96]: Const

Comment (nmasdoufi-ol) [reply agent/exploits/vhd_ptz_exploit.py:96]: understood

**Pull request closed without merging**